### DISCUSSION TITLE
[Question][Hw4] Can someone share the module versions?

### DISCUSSION THREAD

**Discussion by akthddus22** [prosyslab-classroom/cs348-information-security#712] [Q&A]
Name: Songyeon Ma

Hello. I'm struggling with version control. Please, Can somebody help me?
I'm using Opam to control modules and it worked for hw1~3 perfectly, but when i 'make' hw4, it raises an error.
When i make hw4 first time, it raised error 'Library 'cil' not found.' So, I installed cil with 'opam install cil'.
By doing so, I could solve the problem. But another error occured, 'Unbounded module 'Fun'.'
And also, another error 'Unbounded module 'Float'' occured when i tried to make hw3.
By searching, I knew that this error occurs when ocaml version is lower than 4.07.0.
So I tried to upgrade ocaml version to 4.07.0, but for doing that, I needed to delete 'cil' because It works only for ocaml whose version is lower then 4.05.0.
When I downgrade ocaml version to 4.05.0 again, it made same error 'Unbounded module 'Fun'' again.
I struggled with this error for two days... But nothing worked for me...
Please can somebody share me solution or your module versions? Below is my module versions.
Please somebody help me T.T

![envlist](https://github.com/prosyslab-classroom/cs348-information-security/assets/48027227/e41ccacf-94ee-49fa-81ff-3deab5dc4b72)

**Comment by m-spitfire** [comment]
Hello. I suggest you to use provided docker image/KCloud server, or follow the steps to install dependencies required for this assignment from github workflow file.

**Comment by akthddus22** [comment]
> Hello. I suggest you to use provided docker image/KCloud server, or follow the steps to install dependencies required for this assignment from github workflow file.

Thank you for suggestion! let me try...

**Comment by m-spitfire** [comment]
FYI: grading uses ocaml 4.13.1

**Comment by sujin0529** [comment]
We use `https://github.com/prosyslab/cil` in our homework.
So if you are doing a homework in a different environment, you should run this command `opam pin add git+https://github.com/prosyslab/cil`.

Thanks.

**Comment by akthddus22** [comment]
Problem solved by reinstalling whole env. Thanks for all advices!
